presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java

```diff
@@ -1730,6 +1731,20 @@ public Duration getPartitionLeaseDuration()
         return partitionLeaseDuration;
     }
 
+    @Config("hive.partition-lease-renewal-period")
+    @ConfigDescription("Periodically renew partition lease with this frequency in seconds." +
+            "It needs to be set to lower value than hive.partition-lease-duration to renew lease just before expiring.")
```

Comment:
Should we add a check for this ? 
lets say hive.partition-lease-duration = X, hive.partition-lease-renewal-period = Y
1. x=0 & y >0 
2. y>x
3. x>0 & y =0 
what happens  in these scenarios ?

Reply:
Thanks for that thoughtful approach.

```
x=0 & y >0
```
Nothing happens as leasing is effectively disabled. I'm not sure if we should explicitly fail in that case. I don't think so.

```
y>x
```
This is a dangerous situation to be in as the lease may potentially expire. Although it won't be a worse situation than today when there is no lease renewal. 

```
x>0 & y =0
```
This is equivalent to today's state. Fixed lease is acquired at the beginning and never extended.